### DISCUSSION TITLE
'Invalid get index '_room' (on base 'Dictionary')' error in Godot 4

### DISCUSSION THREAD

**Discussion by MeBadDev** [GodotNuts/GodotFirebase#331] [Q&A] [answered]
I was following the wiki, but i got an error when running codes below:
```
func _ready():
	print('logging in')
	Firebase.Auth.signup_failed.connect(login_failed)
	Firebase.Auth.login_anonymous()
	await Firebase.Auth.signup_succeeded
	var request :FirebaseDatabaseReference= await Firebase.Database.get_database_reference("PlayerLevels").get_data()
	print(request)
```

![image](https://user-images.githubusercontent.com/94725241/218704448-8a7ca4c2-233b-4fd0-9904-239f07a5fd1f.png)

Am i doing anything wrong? how should i solve it?

**Comment by WolfgangSenff** [comment, marked as answer]
For one, you shouldn't need to use get_data() in this fashion: when you get a database reference, that will get the data for you. You did not follow the wiki correctly, or the wiki is wrong maybe? Where does it say in the wiki to call get_data() like that? I don't see it anywhere on the page for the database. For two, you should be connecting to the `new_data_update` signal on the reference and then getting your data back from the handler.

**Reply by MeBadDev** [reply]
My bad, I didn't find it in the wiki, but in [another discussion](https://github.com/GodotNuts/GodotFirebase/discussions/322), thanks for answering!
